### ISSUE TITLE
Code Block Background Color only where text is

### ISSUE THREAD

**Issue by phuhl** [jrblevin/markdown-mode#794]
## Expected Behavior

Org mode and most online Markdown visualizations show the background color for code blocks full width:

Org-mode:
![image](https://github.com/jrblevin/markdown-mode/assets/16835755/c0483a17-3c79-4be9-b0b0-c7acd66a659e)

Markdown rendered by Github:
![image](https://github.com/jrblevin/markdown-mode/assets/16835755/08f56f7a-c765-47ac-973b-69fc511cc8f0)

Also, I personally find it easier to work with code in a block that is full width colorized.

## Actual Behavior

Currently markdown mode applies  `markdown-pre-face` only where text is within a code block:

![image](https://github.com/jrblevin/markdown-mode/assets/16835755/916f139e-d40c-4cf1-9b6b-243b44e8a666)

Should this be intended behavior, maybe a feature flag could be added to switch between viewing modes for code blocks.

## Steps to Reproduce

1. Set a background color to  `markdown-pre-face` that is different from the rest:
    ```(custom-set-faces `(markdown-pre-face ((t :background "#ff0000"))))```
2. Create code block in md mode

## Backtrace

n/a

## Software Versions

- Markdown Mode: version 2.6-alpha, 20230817.847 from Melpa
- Emacs: GNU Emacs 29.1.50 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.38, cairo version 1.17.8) of 2023-08-16
- OS: Arch Linux

**Comment by syohex** [comment]
Could you try to use `:extend` attribute as the below sample, please ?

```elisp
(set-face-attribute 'markdown-pre-face nil
                    :foreground "black" :background "white" :weight 'bold
                    :extend t)
```

This is a screenshot that sets `extend` attribute

![image](https://github.com/jrblevin/markdown-mode/assets/554281/2e656a9c-3bd8-445a-8803-0dd691b70e4b)

**Comment by phuhl** [comment]
That solved the issue, thank you!

**Comment by phuhl** [comment]
Interestingly, for code blocks through indentation, this seems to resolve the issue only for every line but the last:
![image](https://github.com/jrblevin/markdown-mode/assets/16835755/3bc1da0e-b6e1-473b-946d-5425740db3dc)

Any idea how this could be resolved?

**Comment by syohex** [comment]
It's impossible with the current implementation. I'm checking whether we can fix that and its side effects

**Comment by syohex** [comment]
I have merged #796. Please check latest version.

**Comment by phuhl** [comment]
Seems like it works. Thanks again!
